controllers/volumereplicationgroup_controller.go

```diff
@@ -27,10 +27,12 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
```

Comment:
Add an import for VolumeReplication shim `volrep "github.com/shyamsundarr/volrep-shim-operator/api/v1alpha1"` and use that instead of adding the entire spec to the `volumereplicationgroup_types.go` file.

I [updated](https://github.com/ShyamsundarR/volrep-shim-operator/commit/2f9f9b83efb60c6e91e9a39a2a1bbf1e60e56289) the shim to name the package more appropriately and so this should onw work.

Reply:
I added the volrep import to the code to find that there is an error in the reconciler:
```
ERRO[0005] no kind is registered for the type v1alpha1.VolumeReplication in scheme "pkg/runtime/scheme.go:101"Failed to get VolumeReplication CR  source="volumereplicationgroup_controller.go:105"
2021-03-12T17:24:06.772-0800	ERROR	controller-runtime.manager.controller.volumereplicationgroup	Reconciler error	{"reconciler group": "ramendr.openshift.io", "reconciler kind": "VolumeReplicationGroup", "name": "volumereplicationgroup-sample-1", "namespace": "default", "error": "failed to get CR: no kind is registered for the type v1alpha1.VolumeReplication in scheme \"pkg/runtime/scheme.go:101\""}
```

I encountered this error before when I forgot to add the VolumeReplication Kind to the Scheme at: https://github.com/priyaranjan-behera/ramen/blob/043f04e83d0ee2a4551f96edc097112a24f4098b/api/v1alpha1/volumereplicationgroup_types.go#L100

I see that the VolRep code does add the kind to the scheme, but somehow the Reconciler code is not able to recognize it.

Reply:
This required addition of `utilruntime.Must(volrep.AddToScheme(scheme))` to init function in main.go. This adds the required scheme to the controller and does the needful.

I took all your commits and squashed them, and further added the above change to PR #11 
Let us know if we can proceed with the other PR (added a signed off by line against your ID as well to the commit message, jFYI)